packages/design-system/src/components/primitives/dnd/geometry-utils.ts

```diff
@@ -0,0 +1,306 @@
+export type Rect = Pick<DOMRect, "top" | "left" | "width" | "height">;
```

Comment:
I don't know why I actually extended DOMRect, could be just a numeric thing, probably won't have compatibility issues

Reply:
Do you mean to just define it as an object instead of using Pick?

Reply:
yeah, if that works in terms of compatibility with DOMRect so one can pass both DOMRect and a React